src/iuse.cpp

```diff
@@ -1841,6 +1841,32 @@ int iuse::purify_iv(player *p, item *it, bool, const tripoint& )
     return it->type->charges_to_use();
 }
 
+void spawn_spores( const player &p ) {
+    int spores_spawned = 0;
+    for ( const tripoint &dest : closest_tripoints_first( 4, p.pos() ) ) {
+        if ( g->m.move_cost( dest ) == 0 ) { // impassable terrain
+            continue;
+        }
+        float dist = trig_dist( dest, p.pos() );
+        if ( x_in_y( 1, dist ) ) { // slightly less frequent than previous code
+            g->m.marlossify( dest );
+        }
+        if ( dest == p.pos() ) { // on top of the player
+            continue;
+        }
+        if ( g->mon_at( dest ) != -1 ) { // there's already a monster there
```

Comment:
Should probably check for `critter_at() != nullptr` (it considers the player (gets rid of the above check) and NPCs).

And some of the comments are ... redundant / pointless:
"impassable terrain" that's *exactly* what the if-condition says.
"slightly less frequent than previous code" soon nobody will know what "previous code" is, and why would it be important to know that it was less or more frequent? This kind of information may go into the commit message.